### PR TITLE
Fix bounded operation outcomes, admission, and checking discovery

### LINKED ISSUE #3467: [Finite-field polynomials] Admit tractable degree-27–108 factorization inputs
## Current limitation and classification

Scale/backend improvement to `polynomial.galois.factor_mod_p.compute`. The operation has the required factorization postcondition, but `MAX_FACTOR_DEGREE = 12` rejects useful polynomials before arithmetic begins. This is an execution-envelope limitation, not incorrect factorization.

Source: [Galois factor request and limits](https://github.com/morluto/jacobian/blob/2b33e8941d2bfa94bca47b1619401c45065366b3/src/jacobian/math/number_theory/galois/_models.py). Local source revision: `2b33e8941d2bfa94bca47b1619401c45065366b3`. Installed MCP/catalog identity was not independently matched to that revision.

## Operation failure record

A live request with `field_order: 3` and ascending coefficients `[1] + [0]*107 + [1]` (the polynomial x^108+1) returned `INVALID_REQUEST`, with operation-validation `too_long`: at most 13 coefficients, received 109. Rejection happened in request validation, not a timeout. The complete raw transport envelope and exact installed operation version were not retained.

This sparse polynomial is a minimal admission reproduction, not one of the irreducible motivating packets.

## Actual motivating packets and backend evidence

The following construction reproduces four polynomial packets from a descending-product divisibility investigation. It was run using the installed python-FLINT backend:

```python
from flint import fmpz_poly, fmpz_mod_poly_ctx
Y = fmpz_poly([0, 1])
B = Y*(Y-1)*(Y-2)
A1 = 2*Y**2*(Y-1)*(Y-2)**2
A2 = 2*Y**2*(Y-1)**2*(Y-2)
R = 1+B*(Y+10)
Z1 = A1**2*A2**3*R**5
Z2 = A1*A2**2*R**3
packets = [(1+Z1, 163), (Z1**2-Z1+1, 5),
           (1+Z2, 79), (Z2**4-Z2**3+Z2**2-Z2+1, 3)]
for f, p in packets:
    g = fmpz_mod_poly_ctx(p)(list(f.coeffs()))
    print(f.degree(), p, g.is_irreducible())
```

All four returned true. Single-run backend-only observations were 0.750 ms (degree 45, p=163), 2.225 ms (degree 90, p=5), 0.236 ms (degree 27, p=79), and 2.879 ms (degree 108, p=3). These exclude transport and parsing and are not a general performance bound. Repeated Frobenius powers and the Rabin gcd conditions agreed; those checks used the same arithmetic backend and are not an independent implementation.

## Desired outcome and bounds

Admit these cases under documented polynomial-degree, modulus-size, coefficient-ingestion, intermediate-work, and output bounds. Separate this envelope from the much smaller degree requirements of Galois-group computation. Existing finite-field presentation limits are not a substitute: constructing an extension of order p^108 is unnecessary for testing a degree-108 polynomial over GF(p).

A maintained FLINT adapter is a possible direction. Retain canonical source-bound factorization results, multiplicities, and reconstruction tests. If irreducibility-only checking warrants its own atomic operation, review that separately rather than requiring full factorization for every decision. Explicitly document ascending coefficient order.

## Acceptance evidence

- Public/native support for the four packets above, with matching semantics.
- Reducible and repeated-factor controls and factor-product reconstruction.
- Correct handling of zero reduction, leading-coefficient degree loss, and nonprime moduli.
- Admission, cancellation, and serialized-output bounds tested at realistic boundaries; resource rejection remains distinct from a mathematical negative.

Related completed work: #2122 addressed multiplicities/irreducibility correctness; #1123 addressed maintained finite-field backends. This request concerns the remaining admitted scale.

### LINKED ISSUE #3468: [Graph partitions] Admit low-work edge-clique certificates above 4096 parts
## Observed behavior

`graph.edge_clique_partition.check` rejects a valid partition of K_92 into its 4,186 individual edges because `parts` is capped at 4,096. Checking that certificate requires only 4,186 pair visits, well below the separately declared 1,000,000-pair work budget.

This is a request-admission improvement, not an incorrect mathematical verdict. Source: [partition models and bounds](https://github.com/morluto/jacobian/blob/2b33e8941d2bfa94bca47b1619401c45065366b3/src/jacobian/math/graphs/clique_partition/_models.py), inspected at `2b33e8941d2bfa94bca47b1619401c45065366b3`. Installed MCP/catalog revision was not independently matched.

## Reproduction

Generate the operation arguments as follows and submit them to `graph.edge_clique_partition.check`:

```python
from itertools import combinations
vertices = [str(i) for i in range(92)]
edges = [list(e) for e in combinations(vertices, 2)]
arguments = {'graph': {'vertices': vertices, 'edges': edges},
             'parts': edges}
```

The live check was rejected by the 4,096-part input limit. The full raw error envelope was not retained. No solver search is needed: each graph edge occurs in precisely its own K_2, so this is a valid certificate by construction.

The motivating research also produced a reported 234-vertex, 8,815-part certificate. That artifact was not replayed in this audit; K_92 is the independently reproduced admission case.

## Expected behavior

Allow valid low-work certificates within the supported graph carrier. Price the request by total vertex references, sum of binomial(part_size, 2), and bounded response size, rather than a part-count cap that excludes ordinary all-K_2 partitions. The current graph envelope can contain substantially more than 4,096 edges.

Update request and result limits coherently. Preserve the source-bound result and existing invalid-partition diagnostics. This request does not require expanding the graph vertex envelope or computing an optimal partition.

## Acceptance evidence

- K_92 with all 4,186 K_2 parts is accepted as valid.
- Larger mixtures of K_2/K_3/K_4 parts are admitted when their charged checking work and output fit.
- Missing edges, repeated coverage, and nonedges still fail with the appropriate mathematical result.
- Oversized malformed requests are rejected before expensive expansion.

Related: #3179 and its implementation #3224 introduced the checker. This is a follow-up on its useful execution envelope, not another checker proposal.

### LINKED ISSUE #3469: [Linear programming] Check supplied rational primal–dual candidates without solving
## Observed mathematical task

A numerical LP solver supplies a candidate, the caller reconstructs its coordinates over the rationals, and an exact check must establish primal feasibility, dual feasibility, and equal objectives. This is useful for fractional covers and weighted combinatorial bounds when finding a candidate is much harder than checking it.

## Missing mathematical result

Given an existing canonical rational LP and supplied primal and dual candidates, determine whether they certify optimality without running optimization again. This is an operation gap in the inspected public surface, not a claim that current solver results are incorrect.

The public operations `optimization.linear.rational_optimum.compute` and `optimization.linear.rational_general_optimum.compute` solve programs. I found no supplied-candidate checking input in their inspected contracts. Result deserialization is structural and should not be repurposed as implicit certificate checking.

Relevant existing implementation: [_certify_point and solver checks](https://github.com/morluto/jacobian/blob/2b33e8941d2bfa94bca47b1619401c45065366b3/src/jacobian/math/optimization/operations.py), inspected at `2b33e8941d2bfa94bca47b1619401c45065366b3`. Reuse the existing LP values and domain-owned arithmetic where appropriate. No nonexistent checker was invoked; this finding comes from catalog and source inspection.

## Existing discovery attempts

`verify exact rational linear programming primal dual optimality` ranked linear-code dual computation first and the general LP solver next. `check supplied primal dual candidates for a rational linear program without solving` also returned unrelated operations and the LP solvers. Search failure alone is not the evidence; the plausible LP contracts were inspected.

## Desired behavior

Accept the source LP plus exact rational candidate data. Explicitly check primal constraints and bounds, dual signs/constraints and bound multipliers, and exact objective equality. Support the source model's minimization/maximization and free-variable conventions. Return a source-bound result that distinguishes a failed candidate from malformed input or an admission failure, with useful residuals or failed conditions.

A negative result means this candidate does not establish optimality, not that the source LP is infeasible. This should be an explicit domain operation, not a generic assurance wrapper or mathematical work in a result validator.

## Bounds and evidence plan

Checking should be admitted by matrix entries, rational digit sizes and predicted arithmetic/intermediate/output work, separately from basis-search limits. The kernel should not invoke simplex to rediscover a supplied solution.

Acceptance examples should cover valid primal–dual pairs, perturbed primal and dual coordinates, equal objectives with infeasible candidates, differing objective directions, free variables, and active bounds. Include a tractable supplied certificate beyond the exact solver's search envelope, chosen during implementation against the current limits. This last case is proposed validation, not a reproduction already run.

## Prior work and disposition

#969 originally included certificate verification and was closed after the exact LP implementation. #2276 addressed source binding/certificate replay. This narrowly scoped follow-up concerns externally supplied candidates without reoptimization. It supports catalog consideration, subject to the normal operation review; it does not prescribe a new generic proof framework.

### LINKED ISSUE #3470: [Discovery] Rank operations by mathematical input and requested checking task
## Observed behavior

Live `math.find` queries returned leading operations whose input contracts do not establish the requested result:

| Query | Observed leading result | Contract mismatch |
|---|---|---|
| `verify finite field irreducibility of a degree 108 polynomial using Frobenius` | `polynomial.galois.frobenius_cycle.compute` | Consumes supplied factorization degrees; does not test the source polynomial. The actual factorization operation appeared next but admits degree at most 12. |
| `verify exact rational linear programming primal dual optimality` | `code.linear.dual.compute` | Computes the dual of a linear code, not an LP certificate. |
| `check polynomial identity over integers` | `polynomial.sos.decomposition.check`, followed by modular polynomial identity computation | SOS checking needs a supplied decomposition; modular identity has a different mathematical domain. |

The queries and returned ordering were observed during the audit. Complete raw response envelopes and installed catalog version identifiers were not retained. Local ranking code was inspected at `2b33e8941d2bfa94bca47b1619401c45065366b3`: [operation field terms and relevance](https://github.com/morluto/jacobian/blob/2b33e8941d2bfa94bca47b1619401c45065366b3/src/jacobian/catalog/search.py). Installed and local revisions were not independently matched.

## Classification and impact

Discovery improvement, not a mathematical correctness bug. Operations may be individually correct while being poor fits for the request. In particular, a cycle derived from supplied degrees is not an irreducibility certificate for the polynomial those degrees supposedly describe.

Other queries worked: Bernstein coefficients and edge-clique partition checking found the intended operations. The goal is targeted regression improvement, not a claim that discovery is generally unusable.

## Desired outcome

Preserve meaningful phrases such as “linear programming” and distinguish computing an object from checking externally supplied candidates. Rank using the relevant mathematical input/result contracts as well as lexical overlap. Make important admitted-scale limits visible in concise discovery metadata so callers can recognize inapplicable results before constructing large payloads.

The current lexical field-term ranking is a plausible improvement point, not proof that one particular ranking algorithm is required. Avoid task-specific aliases for individual conjectures or a strategy/planning layer in the kernel. If no applicable operation exists, neighboring results should not imply that the task is supported.

## Acceptance evidence

Add the three observed queries as regressions with explicit incompatible-result negatives. Test paraphrases, nearby legitimate linear-code and Frobenius-cycle queries, and successful existing Bernstein/partition searches to avoid fixing one phrase by degrading another. Evaluate actual input/result compatibility, not only a hardcoded operation appearing somewhere in a long result list.

Related completed discovery work includes #3235 and #3227. #1747 covers broader held-out composition evaluation; this issue owns the concrete retrieval behavior and contract metadata, rather than duplicating that benchmark project.

### LINKED ISSUE #3471: Convert kernel TimeoutError/InterruptedError to OperationExecutionTimeoutError/Cancelled in counting and augmentation paths
## Summary
Some math kernels emit raw built-in control exceptions (`TimeoutError` / `InterruptedError`) instead of Jacobian execution exceptions (`OperationExecutionTimeoutError` / `OperationExecutionCancelledError`).

Because `execute_operation` and MCP/CLI handlers only special-case domain contract exceptions and `OperationExecution*` exceptions, these raw exceptions bypass the bounded transport error mapping and currently surface as generic tool/CLI failures.

## Files and evidence
- `src/jacobian/math/combinatorics/_counting_process.py:47`
  - raises `TimeoutError("request deadline expired before exact counting")`
- `src/jacobian/math/combinatorics/_counting_process.py:74`
  - raises `TimeoutError("request deadline expired during exact counting")`
- `src/jacobian/math/combinatorics/_counting_process.py:72`
  - raises `InterruptedError("request cancelled during exact counting")`
- `src/jacobian/math/graphs/triangle_free_diameter_augmentation/_augmentation_z3.py:548`
  - raises `TimeoutError("augmentation request expired during no-op presolve")`
- `src/jacobian/math/graphs/triangle_free_diameter_augmentation/_tools.py:19-21`
  - no try/except around call, so these exceptions are passed directly to `Catalog` dispatch.
- `src/jacobian/dispatch.py:113` and `src/jacobian/mcp/tools.py:141-149` only normalize `OperationExecutionTimeoutError` / `OperationExecutionCancelledError`.

## Impact
- Timeout/cancel paths become `ToolError("operation execution failed")` or unstructured CLI exceptions instead of deterministic operation-stage errors.
- Timeout/cancel telemetry is lost and request envelopes cannot communicate precise stage semantics (`operation_execution` vs request parsing) for these operations.

## Reproduction (local)
```python
import sys
sys.path.insert(0, 'src')
import jacobian.math.combinatorics.operations as ops

# Simulate counting worker timeout path
orig = ops.evaluate_count
ops.evaluate_count = lambda operation, n, k: (_ for _ in ()).throw(TimeoutError('simulated'))

try:
    ops.canonical_binomial(1, 0)
except Exception as exc:
    print(type(exc).__name__, str(exc))
finally:
    ops.evaluate_count = orig
```
Output:
`TimeoutError simulated`

A similar monkeypatch against `jacobian.math.graphs.triangle_free_diameter_augmentation._augmentation_z3.solve_triangle_free_diameter_augmentation_values` shows `TimeoutError` escaping.

## Suggested fix
- Replace these raw timeout/cancel exceptions with:
  - `OperationExecutionTimeoutError` in all expired-deadline paths.
  - `OperationExecutionCancelledError` for explicit cancellations.
- Where a contract-preserving non-success result is intended (e.g., augmentation budget exhaustion), return the modeled status (`SOLVER_BUDGET_EXCEEDED`) instead of raising.

### LINKED ISSUE #3473: Response-validation expiry ValueError is misclassified as malformed output
## Summary
Several operations raise `ValueError("request expired during response validation")` when request budgets or cancellation expire during response validation, but these are currently handled in generic malformed-output exception blocks.

This causes timeout-cancel outcomes to be reported as malformed output, which makes triage and telemetry inaccurate.

## Evidence
- src/jacobian/math/graphs/optimization/_invariants.py:481
  - raises ValueError("request expired during response validation")
  - then falls into malformed-output except block
- src/jacobian/math/graphs/optimization/_chromatic_number.py:151
  - same pattern
- src/jacobian/math/combinatorics/finite_structures/hypergraphs/_independence_z3.py:431
  - same pattern

## Impact
- Budgets/cancellations during response validation are indistinguishable from malformed JSON/parsing failures.
- Retry/observability logic may route these cases incorrectly.

## Suggested fix
Prioritize request-expired handling before the generic malformed-output catch in these validation blocks, and route through the standard request-expired/cancel path used elsewhere.

## Note
Issue 3471 covers raw TimeoutError/InterruptedError handling in other kernels; this is the response-validation-specific misclassification variant.

### PR DESCRIPTION
**Pull request by morluto**
## Problem

Response-validation deadline expiry was being reported as malformed output, and several worker adapters leaked raw timeout/cancellation exceptions or treated cancellation as an ordinary solver outcome. Separately, fixed degree/part-count limits rejected inexpensive mathematical checks, discovery favored operations with the wrong input or task, and checking supplied rational LP candidates required solving instead.

Closes #3473
Closes #3471
Closes #3470
Closes #3469
Closes #3468
Closes #3467

## Outcome

- Preserve typed timeout/cancellation outcomes through counting, augmentation, graph/hypergraph optimization, isomorphism, maximum-cut, and unsat-core worker paths. Expiry during response validation takes precedence over malformed-output classification; operations with an existing bounded unknown outcome retain that contract.
- Admit low-work edge-clique certificates beyond 4096 parts using vertex-reference and pair-work bounds, with result cardinality bounded by the source carrier.
- Factor the motivating degree-27–108 finite-field inputs using maintained deterministic SymPy routines in a killable worker, without symbolic expression construction. Retain explicit degree, field, and work bounds.
- Publish `optimization.linear.rational_optimality.check`: check supplied primal feasibility, dual signs/stationarity, and equal objectives using exact rational arithmetic, without invoking a solver or expanding to a slack-variable formulation.
- Add modest mathematical-phrase ranking and respect explicitly requested coefficient domains for checking tasks. Unsupported integer-polynomial identity checking is not presented as an available capability.
- Address related scale defects: use FLINT sparse polynomial arithmetic for SOS/Gram reconstruction and bound graph symmetry by actual generator actions rather than a fixed 4096-edge ceiling.

Existing LP solving operations remain available and unchanged; no operation ID is superseded. This does not remove exponential-search limits or claim arbitrary-degree factorization.

## Validation

Review follow-up head `dffaa7919` addresses the three review threads: complete compute operations remain discoverable for checking queries; partition cardinality is checked before nested parsing; and finite-field backend calls run in a killable worker under the shared request deadline. The earlier-deadline and cancellation tests stop and reap children inside both backend phases and reject stops during setup without launching a worker. Validation passed: 63 affected math tests, 143 catalog tests, 6 process regressions, 306 tooling tests, 908 catalog integration tests, full lint, mypy, architecture, repository hygiene, import contracts, and build/wheel-content checks. Hosted CI is rerunning on this head.

CI architecture follow-up `f8195858c` separates the canonical LP candidate from the wire request, shares partition structural validation without constructing request models, and replaces serialization-byte admission with mathematical cardinality/digit bounds. Full lint, type checking, architecture, repository hygiene, TODO checks, import contracts, documentation links, npm tests, and the package build/wheel-content check passed locally. The repair passed 48 focused mathematical regressions, 141 catalog tests, and 908 catalog integration tests; the final structural-validation repair was covered by the focused rerun. Hosted CI is rerunning on this head.

Evidence covers the tree committed as `de8aada6d3e797e9cf0dea3fbc53cc9ccfd345a8`. The full math run preceded final focused repairs; affected owner tests were rerun after those repairs.

- `make affected AFFECTED_BASE=origin/main`: static checks and full math lane completed, with **8,924 math tests passing**. The initial catalog snapshot failure was repaired, and remaining lanes were completed separately rather than repeating the full math lane.
- `make test-catalog`: **141 passed**.
- `make test-integration`: **180 passed**; `make test-integration TESTS=tests/integration/catalog/`: **908 passed**.
- CLI, dispatch, tooling, and MCP lanes: **4**, **129**, **306**, and **70** passed respectively.
- Process lane: **296 passed** with one obsolete timeout expectation; after updating that expectation, `make test-process TESTS=tests/process/test_graph_isomorphism_worker.py`: **3 passed**.
- `make test-singular test-qepcad`: passed; QEPCAD reported **27 passed**.
- Focused LP/discovery regressions: **31 passed**; affected SMT tests: **111 passed**.
- Changed-file Ruff and mypy checks and `git diff --check` passed.
- Motivating factorization, partition, discovery, sparse SOS, and symmetry rejections were reproduced before repair. Factorization regressions independently check irreducibility with FLINT and reconstruct products.
- Hosted CI, including installed-wheel validation, is pending.

## Contract impact

Public impact: one new LP-check operation and native entry point; wider bounded factorization, Frobenius-degree, partition, sparse SOS/Gram, and symmetry inputs; corrected operational failure classification; discovery ranking changes.

- [x] Cheap malformed or over-budget input is rejected before expensive work.
- [x] Exact results retain canonical mathematical values and reconstruction context.
- [x] Native and public execution paths have relevant acceptance and rejection regressions.
- [x] Producer-to-serialization-to-consumer composition and invalid supplied claims are covered.

Structural schemas describe carrier limits; semantic work, rational-growth, and output limits remain request admission checks, not claims that JSON Schema alone proves mathematical admissibility.

## Review closure

- [x] Relevant validation and specialist lanes are listed above; checks invalidated by repairs were rerun.
- [x] The intended base was fetched before pushing and had not advanced.
- [x] No compatibility or shadow paths were added.
- [ ] Hosted CI evidence matches the final head SHA.

Suggested review order follows the six commits: execution outcomes, partition admission, factorization, LP checking, discovery, then related sparse/symmetry scale repairs.

### Operation boundary

Admission remains domain-owned and precedes backend expansion:

- Partition checking: up to 32,640 parts, 65,280 vertex references, 1,000,000 pair visits, and at most 32,640 covering indices.
- Finite-field factorization: degree at most 128, prime field order at most 251, and deterministic work estimate `32 * (p + 1) * d^3 <= 1,000,000,000`; bounded quadratic storage and a shared 60-second safety deadline.
- LP checking: existing canonical standard/general rational LP values; at most 32 variables, 64 rows, 128-digit candidate scalars, 20,000 scalar updates, 32,768 predicted intermediate digits, and 8,388,608 retained scalar digits, under a shared 60-second deadline.
- SOS/Gram: sparse support and coefficient-height admission remain; canonical exponent limits replace the degree-12 ceiling. Returned squared summands stay within the canonical exponent carrier.
- Symmetry: up to 64 generators, 278,528 generator-action entries, and at most one orbit record per vertex or edge.

No generic verification wrapper or request-persistent state is introduced. The existing process supervisor also isolates finite-field square-free decomposition and Berlekamp splitting; response construction does not rerun factorization.

### Canonical value audit

LP checking reuses existing canonical rational and program types, retains the authored source and candidates, and reports exact objectives, residuals, and failed conditions. Feasible primal/dual candidates with equal objectives establish optimality; a failed candidate does not establish infeasibility. Regressions cover wrong signs, absent/free bounds, perturbed candidates, equal-objective infeasible candidates, and serialization.

Factorization retains units and multiplicities with product reconstruction. Partition tests cover complete-graph certificates through 256 vertices and invalid coverage. Sparse polynomial tests include high-degree, multi-axis, invalid-identity, and output-boundary cases.

### Catalog admission

The new LP operation establishes one reusable postcondition: whether supplied rational primal/dual candidates establish optimality for the retained source program. Existing solver operations cannot provide this without solving, and solver normalization can exceed their admission envelope even when candidate checking is cheap.

The public example and discovery-selected operation are exercised on supplied valid and invalid candidates. Exact fraction arithmetic checks source-coordinate relations directly; no presolve, basis enumeration, or solver normalization determines admission. Fixed row, variable, and scalar caps are explicit bounded-carrier limits, supplemented by work, growth, and output admission.

### CODE REVIEW

File: src/jacobian/catalog/search.py [hunk 1]
```diff
@@ -309,6 +314,76 @@ def token_set(value: str) -> frozenset[str]:
...
+    if normalize_discovery_term(domain.group(1)) not in terms:
+        return False
+    if not frozenset().union(*fields[:2]) & {
+        "check",
+        "verify",
+        "test",
+        "decide",
+        "compare",
+    }:
+        return False
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/catalog/search.py]: Fixed in 5a5073c24.

Removed the declaration-verb gate: a complete compute operation remains eligible when it establishes the requested mathematical check. Explicit coefficient-domain and mathematical-object matching remain in place. The reported multivariate-factorization query and a checking paraphrase now rank polynomial.multivariate.factor.compute first; all 143 catalog tests pass.

File: src/jacobian/math/number_theory/galois/_factor.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    with request_execution(time.monotonic()) if execution is None else nullcontext():
+        execution = current_request_execution()
+        assert execution is not None
+        deadline = execution.started_at + FACTOR_WALL_SECONDS
+        if execution.deadline is not None:
+            deadline = min(deadline, execution.deadline)
+        bind_request_deadline(deadline)
+        request_checkpoint("before finite-field factorization")
+        result = _factor_polynomial(prime, coefficients)
+        request_checkpoint("after finite-field factorization")
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/galois/_factor.py]: Fixed in dffaa7919.

Moved deterministic square-free decomposition and Berlekamp splitting into an owner-local worker supervised by the existing bounded process runner. The caller/owner deadline covers setup, execution, decoding, and cleanup; startup recomputes the remaining allowance after directory creation. Typed timeout/cancellation bypass the OSError startup handler. Six process regressions cover both backend phases, child reaping, and stops during setup. The motivating factorization packets and reconstruction tests remain accepted.

File: src/jacobian/math/graphs/clique_partition/_models.py [hunk 3]
```diff
@@ -34,7 +42,7 @@ class EdgeCliquePartitionRequest(StrictModel):
     """
 
     graph: SimpleUndirectedGraph
-    parts: tuple[tuple[str, ...], ...] = Field(
+    parts: tuple[PartitionPart, ...] = Field(
         max_length=MAX_PARTITION_PARTS,
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/graphs/clique_partition/_models.py:46]: Fixed in c9b039ce2.

Added a raw before-validator that checks part count, per-part length, and aggregate vertex references before canonicalization or nested label parsing. Native admission also rejects oversized outer containers before traversal and aggregate work before member validation. Regressions cover Python/JSON raw inputs, malformed nested labels in over-budget payloads, and outer tuples that must not be iterated. Valid K92/K134/K256 certificates remain accepted.
